lms/djangoapps/ccx/views.py

```diff
@@ -345,6 +363,14 @@ def set_grading_policy(request, course, ccx=None):
     override_field_for_ccx(
         ccx, course, 'grading_policy', json.loads(request.POST['policy']))
 
+    # using CCX object as sender here.
+    responses = SignalHandler.course_published.send(
+        sender=ccx,
+        course_key=CCXLocator.from_course_locator(course.id, unicode(ccx.id))
+    )
+    for rec, response in responses:
+        log.info('Signal fired when course is published. Receiver: %s. Response: %s', rec, response)
+
     url = reverse(
         'ccx_coach_dashboard',
         kwargs={'course_id': CCXLocator.from_course_locator(course.id, ccx.id)}
```

Comment:
it should be, but CCXLocator with non unicode also works, it only give issues when we compare same CCXLocator one with ccx.id unicode and one with not.

let me fix through out views